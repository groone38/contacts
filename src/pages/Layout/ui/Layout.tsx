import React from "react";
import { Outlet } from "react-router-dom";
import classes from "./Layout.module.scss";
import { Header } from "src/widgets/Header";
import { Footer } from "src/widgets/Footer";

const Layout = () => {
  return (
    <div className={classes.wrapper}>
      <Header />
      <main>
        <Outlet />
      </main>
      <Footer />
    </div>
  );
};

export default Layout;
