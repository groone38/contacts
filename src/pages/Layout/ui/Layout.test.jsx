import { render, screen } from "@testing-library/react";
import Layout from "./Layout";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";

const FakeComponent = () => <div>fake text</div>;

describe("Layout component", () => {
  it("render", async () => {
    render(
      <MemoryRouter initialEntries={["/"]}>
        <Layout />
      </MemoryRouter>
    );

    expect(screen.getByText("Create new contact")).toBeInTheDocument();
  });
  it("Register snapchot", () => {
    const layout = render(
      <MemoryRouter initialEntries={["/"]}>
        <Layout />
      </MemoryRouter>
    );
    expect(layout).toMatchSnapshot();
  });
});
