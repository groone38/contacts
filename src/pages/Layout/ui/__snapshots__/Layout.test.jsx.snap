// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Register snapchot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <header
          class="header"
        >
          <div
            class="logo"
          >
            <a
              class="link"
              href="/"
            >
              HOME
            </a>
          </div>
          <nav
            class="nav"
          >
            <ul>
              <li>
                Create new contact
              </li>
              <li>
                Exit
              </li>
            </ul>
          </nav>
          <div
            class="burger"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 64c0-17.7-14.3-32-32-32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32zm0 256c0-17.7-14.3-32-32-32H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H416c17.7 0 32-14.3 32-32zM0 192c0 17.7 14.3 32 32 32H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H32c-17.7 0-32 14.3-32 32zM448 448c0-17.7-14.3-32-32-32H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H416c17.7 0 32-14.3 32-32z"
              />
            </svg>
            <div
              class="menu close"
            >
              <ul>
                <li>
                  <a
                    href="/"
                  >
                    HOME
                  </a>
                </li>
                <li>
                  <p>
                    CREATE NEW CONTACT
                  </p>
                </li>
                <li>
                  EXIT
                </li>
              </ul>
            </div>
          </div>
        </header>
        <main />
        <footer
          class="footer"
        >
          <h2>
            © LisinDev
          </h2>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <header
        class="header"
      >
        <div
          class="logo"
        >
          <a
            class="link"
            href="/"
          >
            HOME
          </a>
        </div>
        <nav
          class="nav"
        >
          <ul>
            <li>
              Create new contact
            </li>
            <li>
              Exit
            </li>
          </ul>
        </nav>
        <div
          class="burger"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 64c0-17.7-14.3-32-32-32H32C14.3 32 0 46.3 0 64S14.3 96 32 96H416c17.7 0 32-14.3 32-32zm0 256c0-17.7-14.3-32-32-32H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H416c17.7 0 32-14.3 32-32zM0 192c0 17.7 14.3 32 32 32H416c17.7 0 32-14.3 32-32s-14.3-32-32-32H32c-17.7 0-32 14.3-32 32zM448 448c0-17.7-14.3-32-32-32H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H416c17.7 0 32-14.3 32-32z"
            />
          </svg>
          <div
            class="menu close"
          >
            <ul>
              <li>
                <a
                  href="/"
                >
                  HOME
                </a>
              </li>
              <li>
                <p>
                  CREATE NEW CONTACT
                </p>
              </li>
              <li>
                EXIT
              </li>
            </ul>
          </div>
        </div>
      </header>
      <main />
      <footer
        class="footer"
      >
        <h2>
          © LisinDev
        </h2>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
